src/diagnostics/test/CMakeLists.txt

```diff
@@ -11,14 +11,23 @@ project( diagnostics_test CXX )
 # ---------------------------------------------------------------------------- #
 # Source files
 # ---------------------------------------------------------------------------- #
-file( GLOB test_sources *.cc )
+set(scalar_test_sources
+    ${PROJECT_SOURCE_DIR}/tstDiagnostics.cc
+    ${PROJECT_SOURCE_DIR}/tstProcmon.cc
+    ${PROJECT_SOURCE_DIR}/tstTiming.cc)
+set(parallel_test_sources
+    ${PROJECT_SOURCE_DIR}/tstruntime_error.cc)
 
 # ---------------------------------------------------------------------------- #
 # Build Unit tests
 # ---------------------------------------------------------------------------- #
 add_scalar_tests(
-   SOURCES "${test_sources}"
+   SOURCES "${scalar_test_sources}"
```

Comment:
Just to be clear, `Lib_diagnostics --> MPI`, so even though this is registered as a scalar test, the binary still has links to MPI.  As long as it passes, I'm okay with this change.  Let's see what the CI test report looks like.